overlord/snapstate/snapmgr.go

```diff
@@ -551,6 +556,45 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// ensureSnapConfineApparmor ensures that we have a valid apparmor
+// profile for snap-confine when we are in re-exec mode
+func (m *SnapManager) ensureSnapConfineApparmor() error {
+	if !release.OnClassic {
+		return nil
+	}
+
+	root := filepath.Join(dirs.SnapMountDir, "/core/current/")
+	snapConfineInCore, err := filepath.EvalSymlinks(filepath.Join(root, "/usr/lib/snapd/snap-confine"))
+	if err != nil {
+		return err
+	}
+
+	apparmorProfilePath := filepath.Join(dirs.SystemApparmorDir, strings.Replace(snapConfineInCore[1:], "/", ".", -1))
+	if osutil.FileExists(apparmorProfilePath) {
+		return nil
+	}
+
+	// FIXME2: cleanup old entries here
+
+	// FIXME: what to do for debian? debian has a less powerful apparmor
+	//        so we can not use the ubuntu profile?
+	apparmorProfile, err := ioutil.ReadFile(filepath.Join(root, "/etc/apparmor.d/usr.lib.snapd.snap-confine"))
```

Comment:
I would prefer to do that in a follow-up branch, mostly because AIUI we currently ship a working `/etc/apparmor.d/usr.lib.snapd.snap-confine` *only* on Ubuntu. We also ship it on Debian but AIUI its broken there. And the other distros do not use it (yet). So it seems ideal to do a followup each time we add this file for a new distro (unless my premise is wrong of course).